ditto/metrics/network_analysis.py

```diff
@@ -396,25 +399,55 @@ def export(self, *args):
             card.loc[n_row] = [key] + [data[x] if x in data else None for x in cols[1:]]
             n_row += 1
 
-        # Instanciate the writer
-        xlsx_writer = pd.ExcelWriter(export_path)
-        # Write to excel
-        card.to_excel(xlsx_writer)
-        # Save
-        xlsx_writer.save()
+        # Write to csv
+        card.to_csv(export_path, header=True, index=False)
 
     def tag_objects(self):
         """
         Loop over the objects and fill the feeder_name and substaation_name attributes.
         """
         for obj in self.model.models:
-            if hasattr(obj, "feeder_name"):
+            if hasattr(obj, "feeder_name") and hasattr(obj, "name"):
```

Comment:
Excessive use of `hasattr` makes it hard to know what code path will be taken during execution/runtime. Refactoring the function to make the body of functions smaller may help. Ideally, a function should do just one action. For example, this particular function `tag_objects` could be refactored into `_tag_substation_node`, `_tag_substation_connecting_element`, `_tag_substation_from_element`.